redbox-core/redbox/app.py

```diff
@@ -22,15 +34,40 @@ def __init__(
         all_chunks_retriever: VectorStoreRetriever | None = None,
         parameterised_retriever: VectorStoreRetriever | None = None,
         metadata_retriever: VectorStoreRetriever | None = None,
+        embedding_model: Embeddings | None = None,
         env: Settings | None = None,
         debug: bool = False,
     ):
         _env = env or Settings()
+
+        # Retrievers
+
         _all_chunks_retriever = all_chunks_retriever or get_all_chunks_retriever(_env)
         _parameterised_retriever = parameterised_retriever or get_parameterised_retriever(_env)
         _metadata_retriever = metadata_retriever or get_metadata_retriever(_env)
+        _embedding_model = embedding_model or get_embeddings(_env)
+
+        # Tools
+
+        search_documents = build_search_documents_tool(
+            es_client=_env.elasticsearch_client(),
+            index_name=f"{_env.elastic_root_index}-chunk",
+            embedding_model=_embedding_model,
+            embedding_field_name=_env.embedding_document_field_name,
+            chunk_resolution=ChunkResolution.normal,
+        )
+
+        tools: dict[str, StructuredTool] = {
```

Comment:
just for my understanding - here we have a dict of tools which right now includes just one tool: doc-search, but could be extended to wikipedia, lex, etc?